### DISCUSSION TITLE
How to create an invoice

### DISCUSSION THREAD

**Discussion by andrelfo2021** [Intechnity-com/OdooJsonRpcClient#68] [Q&A]
Hello, I'm trying to integrate the creation of an invoice in Odoo from my net core 5 app. I managed to create the invoice and the lines , but the invoice does not appear in the list of invoices. Please does anyone have the solution or suggestions for this implementation?
Thanks in advance

**Comment by coleblarson** [comment]
I discovered that you need to do these things:

1) make sure the invoice model includes all of the required fields defined and create it first.  (go through the model and look for required: true)

OdooInvoice = OdooDictionaryModel<AccountMoveOdooModel>.Create(() => new AccountMoveOdooModel
            {
                PartnerId = ServiceContractProperty.OdooId,
                Date = DateInvoiced,
                InvoiceDate = DateInvoiced,
                InvoiceDateDue = DateDue,
                Ref = ServiceContractProperty.Property.Address,
                State = StatusAccountMoveOdooEnum.Draft,
                MoveType = TypeAccountMoveOdooEnum.CustomerInvoice,
                IsStorno = false,
                AutoPost = AutoPostAccountMoveOdooEnum.No,
                CurrencyId = 1,
                ExtractState = ExtractStateAccountMoveOdooEnum.WaitingValidation,
                
            });

2.) set the move_id in each lines model to the ID of the invoice you just created, then create all this lines.

I created the lines into a list of dictionaries: List<OdooDictionaryModel<AccountMoveLineOdooModel>> lines
then for each line, set
line["move_id"] = moveId;   //moveId is the Id of the invoice., just before you create it.

If you want a line not to be taxed, set the tax_ids to null.  If you want them to be set to the default, leave the definition out.  Otherwise set the tax_ids to the rates you want...
